test/test_utils/__init__.py

```diff
@@ -1623,40 +1623,18 @@ def get_python_version_from_image_uri(image_uri):
     python_version = python_version_search.group()
     return "py36" if python_version == "py3" else python_version
 
-def construct_buildspec_path(dlc_path, framework_path, buildspec, framework_version):
-    """
-    Construct a relative path to the buildspec yaml file by iterative checking on the existence of
-    a specific version file for the framework being tested. Possible options include:
-    [buildspec-[Major]-[Minor]-[Patch].yml, buildspec-[Major]-[Minor].yml, buildspec-[Major].yml, buildspec.yml]
-    :param dlc_path: path to the DLC test folder
-    :param framework_path: Framework folder name
-    :param buildspec: buildspec file name
-    :param framework_version: default (long) framework version name
-    """
-    if framework_version:
-        # pattern matches for example 0.3.2 or 22.3
-        pattern = r"^(\d+)(\.\d+)?(\.\d+)?$"
-        matched = re.search(pattern, framework_version)
-        if matched:
-            constructed_version = ""
-            versions_to_search = []
-            for match in matched.groups():
-                if match:
-                    constructed_version = f'{constructed_version}{match.replace(".","-")}'
-                    versions_to_search.append(constructed_version)
-
-            for version in reversed(versions_to_search):
-                buildspec_path = os.path.join(dlc_path, framework_path, f"{buildspec}-{version}.yml")
-                if os.path.exists(buildspec_path):
-                    return buildspec_path
-        else:
-            raise ValueError(f"Framework version {framework_version} was not matched.")
 
-    buildspec_path = os.path.join(dlc_path, framework_path, f"{buildspec}.yml")
-    if not os.path.exists(buildspec_path):
-        raise ValueError('Could not construct a valid buildspec path.')
+def construct_buildspec_path(dlc_path):
+    """
+    Construct a relative path to the buildspec yaml file by checking for config override or env variable.
+    """
+    buildspec_file_path = config.get_buildspec_override_from_testjob() or os.getenv("FRAMEWORK_BUILDSPEC_FILE")
```

Comment:
I imagine this may have trouble in mainline context if this env variable is not set.

Reply:
note: if preferred, we can probably just remove this function altogether since it is only used in test cuda paths, and put these lines directly into the test.